spec/foxsi_ospex.pro

```diff
@@ -0,0 +1,198 @@
+
+; sum signal over three strips for each good event
+if keyword_set(three) then spec = make_spectrum( dat[i3], bin=bin,/three,split_limit=split_limit) $
+        else spec = spec
+
+; energy range for counts spectrum
+i = where( spec.energy_kev gt 3. and spec.energy_kev lt 15. )
+enm = spec.energy_kev[i]
+en_lo = enm - deriv(enm)*0.5
+en_hi = enm + deriv(enm)*0.5
+en2 = transpose( [[en_lo],[en_hi]] )
+
+eff = total(spec.spec_p[i])/total(spec_all.spec_p[i])   ;ratio of good events to all events in time range (p-side)
+;eff_n = total(spec.spec_n[i])/total(spec_all.spec_n[i]) 	;ratio of good events to all events in time range (n-side) 
+
+; create response matrix
+resp = get_foxsi_effarea( energy_arr=enm, module=det, offaxis_angle=offaxis, $
+	let_file=let_file, type=type, _extra = _extra )
+max_EA = max(resp.eff_area_cm2[where(resp.eff_area_cm2 ge 0.)])	;determine maximum effective area
+;normalize effective area and multiply by ratio of good to all events
+diag = resp.eff_area_cm2 * eff / max_EA	
+ndiag = n_elements( diag )	
+nondiag = fltarr(ndiag,ndiag)	;create array for nondiagonal response
+
+for j=0, ndiag-1 do nondiag[j,j]=diag[j]	;set diagonal of nondiag to the diagonal response (cm^2)
+sigma = fwhm / 2.355		; energy resolution
+toty = total(gaussian(findgen(ndiag),[0.3989*bin/sigma,round((12./(2*bin))-1.),sigma/bin] ))
+        ; toty is a sum of the values of a normalized Gaussian function binned according to the set 'bin' keyword.
+        ; The standard deviation for the Gaussian is normalized to the bin size (sigma/bin).
+        ; toty should be equal to one, but is slightly off due to rounding.
+
+; compute the nondiagonal response by convolving diagonal response with a Gaussian
+for j=0, ndiag-1 do begin
+	y = diag[j]*gaussian( findgen(ndiag), [0.3989*bin/sigma,j,sigma/bin] )/toty
+                ; y is the convolution of the diagonal response value diag[j] with a
+                ; Gaussian function of standard deviation sigma/bin.
+                ; The values are normalized using toty (described above).
+	nondiag[*,j] = y
+endfor
+
+;create livetime array
+livetime_array = fltarr(ndiag)
+livetime_array[*] = ltime  
+
+; run ospex for spectral analysis
+o = ospex(/no_gui)
+o -> set, spex_data_source = 'SPEX_USER_DATA'
+o -> set, spectrum = spec.spec_p[i]*bin, spex_ct_edges = en2, errors = spec.spec_p_err[i]*bin,livetime=livetime_array
+o -> set, spex_respinfo = nondiag/bin
+o -> set, spex_area = max_EA
+
+; optically thin thermal plasma model (isothermal)
+o-> set, fit_function= 'vth'	
+o-> set, fit_comp_params= [0.003, 1.0, 1.00000]                                
+o-> set, fit_comp_free_mask= [1B, 1B, 0B]
+o-> set, fit_comp_spectrum= ['full', '']
+o-> set, fit_comp_model= ['chianti', '']
+o-> set, spex_erange = erange
+o-> set, spex_fit_time_interval = [0.,3600.]
+
+o-> set, spex_fit_manual=0
+o-> set, spex_autoplot_enable=1
+
+o-> dofit
+o-> savefit, outfile = filename+'.fits'
+
+save, eff, ltime, filename=filename+'.sav'
+
```

Comment:
add 
IF keyword_set(stop) THEN stop

So that one can look at the plots before the object gets cleaned up